runtime/common/src/slots.rs

```diff
@@ -376,6 +403,14 @@ impl<T: Config> Leaser for Module<T> {
 			}
 
 			let reserved = maybe_additional.unwrap_or_default();
+
+			// Check if current lease period is same as period begin, and onboard them directly.
+			// This will allow us to support onboarding new parachains in the middle of a lease period.
+			if current_lease_period == period_begin {
+				let res = T::Registrar::make_parachain(para);
+				debug_assert!(res.is_ok());
```

Comment:
let's always have a proof with any (debug) assertions.

Reply:
hmm, actually Slots does not _guarantee_ that they are a para and that `make_parachain` will succeed.

The `debug_assert` is more to ensure that all tests take into account that this should be successful in the tests, and if not, maybe the test itself is bad.

Am I using this kind of pattern wrong?